src/update-feeds/update-feeds.ts

```diff
@@ -152,52 +151,34 @@ export const decodeBeaconValue = (encodedBeaconValue: string) => {
   return decodedBeaconValue;
 };
 
-export const getFeedsToUpdate = (
+// https://github.com/api3dao/airnode-protocol-v1/blob/fa95f043ce4b50e843e407b96f7ae3edcf899c32/contracts/api3-server-v1/DataFeedServer.sol#L132
+export const encodeBeaconValue = (numericValue: string) => {
+  const numericValueAsBigNumber = ethers.BigNumber.from(numericValue);
+
+  return ethers.utils.defaultAbiCoder.encode(['int256'], [numericValueAsBigNumber]);
+};
+
+export const getFeedsToUpdate = async (
   batch: ReadDapiWithIndexResponse[],
-  deviationThresholdCoefficient: DeviationThresholdCoefficient
-): UpdateableDapi[] =>
-  batch
-    .map((dapiInfo): UpdateableDapi | null => {
-      const beaconsSignedData = dapiInfo.decodedDataFeed.beacons.map((beacon) => getStoreDataPoint(beacon.beaconId));
-
-      // Only update data feed when we have signed data for all constituent beacons.
-      if (beaconsSignedData.some((signedData) => !signedData)) return null;
-
-      const beaconsDecodedValues = beaconsSignedData.map((signedData) => decodeBeaconValue(signedData!.encodedValue));
-      // Only update data feed when all beacon values are valid.
-      if (beaconsDecodedValues.includes(null)) return null;
-
-      // https://github.com/api3dao/airnode-protocol-v1/blob/fa95f043ce4b50e843e407b96f7ae3edcf899c32/contracts/api3-server-v1/DataFeedServer.sol#L163
-      const newBeaconSetValue = calculateMedian(beaconsDecodedValues.map((decodedValue) => decodedValue!));
-      const newBeaconSetTimestamp = calculateMedian(
-        beaconsSignedData.map((signedData) => ethers.BigNumber.from(signedData!.timestamp))
-      )!.toNumber();
-      const adjustedDeviationThresholdCoefficient = multiplyBigNumber(
-        dapiInfo.updateParameters.deviationThresholdInPercentage,
-        deviationThresholdCoefficient
-      );
-      if (
-        !checkUpdateConditions(
-          dapiInfo.dataFeedValue.value,
-          dapiInfo.dataFeedValue.timestamp,
-          newBeaconSetValue,
-          newBeaconSetTimestamp,
-          dapiInfo.updateParameters.heartbeatInterval,
-          adjustedDeviationThresholdCoefficient
-        )
-      ) {
-        return null;
-      }
+  deviationThresholdCoefficient: DeviationThresholdCoefficient,
+  providerName: ProviderName,
+  chainId: ChainId
+): Promise<UpdatableDapi[]> => {
+  const shallowCheckedDapis = shallowCheckFeeds(batch, deviationThresholdCoefficient);
```

Comment:
Doing a shallow check filtering first and afterwards doing the deep filter will not work. Consider this case:
1. Say we have all data for a beacon set update but the aggregated value does not update the feed, but some of our signed data is outdated.
2. The `shallowCheckFeeds` will omit this beacon set.
3. However, it is still possible that the beacon set needs an update (when we replace the outdated signed data values with on-chain ones).

This needs to be implemented in one function, roughly like this:
1. Get all the beacons for all dAPIs in the batch and multicall their values (these should be de-duped by beacon ID).
2. Check update conditions for each dAPI separately.
3. For each beacon use the more up to date option (on-chain, off-chain) and do the aggregation and update check.
4. If the dAPI needs an update, only update the beacons for which you used the off-chain values.

Reply:
I think this is now implemented as required 👌 
LMK if I need to handle a multicall failure with a fallback 🤷 🙏